src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+import {Position} from './Position'
+
+export class Matrix{
```

Comment:
This class is very long, and there's a lot of strategic logic that appears to have nothing to do with `Matrix`. I think you could cut out some of the strategy into separate classes.